django_lti_tool_provider/models.py

```diff
@@ -57,3 +57,36 @@ def send_lti_grade(self, grade):
         ))
 
         return outcome
+
+    @classmethod
+    def get_or_create_by_parameters(cls, user, authentication_manager, lti_params):
+        """
+        Filters out OAuth parameters
+        """
+        custom_key = authentication_manager.vary_by_key(lti_params)
+
+        # implicitly tested by test_views
+        if custom_key is None:
+            custom_key = ''
+
+        lti_user_data, created = LtiUserData.objects.get_or_create(user=user, custom_key=custom_key)
+
+        if lti_user_data.edx_lti_parameters.get('user_id', lti_params['user_id']) != lti_params['user_id']:
+            # TODO: not covered by test
+            message = u"LTI parameters for user found, but anonymous user id does not match. "
+            _logger.error(message)
+            raise ValueError(message)
```

Comment:
Can't we simply log out any other user than the expected one?  This could happen e.g. when two different edx instance access the same Dalite instance, or when two users share a browser, or in any number of other weird scenarios.  I don't see any reason why we should completely disallow this.